test/e2e/clusterctl_upgrade.go

```diff
@@ -156,7 +157,7 @@ type ClusterctlUpgradeSpecInput struct {
 //	- mkdir -p /opt/cluster-api
 //	- aws s3 cp "s3://${S3_BUCKET}/${E2E_IMAGE_SHA}" /opt/cluster-api/image.tar
 //	- ctr -n k8s.io images import /opt/cluster-api/image.tar # The image must be imported into the k8s.io namespace
-func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpgradeSpecInput) {
+func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpgradeSpecInput, machineListGroupVersion schema.GroupVersion) {
```

Comment:
Let's not add it to the function parameters directly, but to the `ClusterctlUpgradeSpecInput` struct instead :-)

Also: let's default it to `v1beta1` if not set.

Using v1alpha3 here is because in the past we had tests to upgrade from e.g. `v0.3` which did not have v1beta1 yet. nowadays the lowest version is v1.0 (at last for the tests of core CAPI). So we can default to v1beta1 and have the parameter to use a different grouip version if required.

Reply:
Just to be clear on: 

> Also: let's default it to v1beta1 if not set.

We would still want to pass v1alpha4 GroupVersion to >=v1.5 tests or leave it out empty (GroupVersion: nil) and let it default to v1beta1?

Reply:
> When testing for CAPI >= v1.5 we will need to at least use v1alpha4 (and we will hit the same with v1alpha4 as soon as it does not get served anymore in a later minor release).

> We may still want to use v1alpha3 for the 1.0=>current and 1.4=>current cases though.

we need to still set it to v1alpha4 I guess and have a default value in case it is not passed in the future :)

Reply:
Should be fine to stick at the default for our defined tests (`v1beta1`), but also okay to pass `v1alpha4` for them.

We just need to make sure it is an APIVersion which is available. so `v1beta1` would also be valid for all our configured tests.

(if we would still have a 0.3 => latest test, we would need to configure the variable, but we don't have.)

By having the parameter we make sure if someone uses the test they are able to configure the lower apiversion if it is required.

cc @sbueringer

Reply:
Doesn't matter that much. But I would say we should use the storage version used by our "old" CAPI version. Which is v1beta1 for >= v1.2

Reply:
So basically we should not pass in the GroupVersion into any of our tests. This is just for folks who still test v1alpha3 or v1alpha4 => main